src/ThriftSQL/Utils/Iterator.php

```diff
@@ -20,34 +22,49 @@ class Iterator implements \Iterator {
   private $buffer;
   private $location;
 
-  private $runCount = 0;
+  private $queryCount = 0;
+  private $rewindCount = 0;
   private $allowRerun = false;
+  private $useSchema = false;
 
   public function __construct( \ThriftSQL $thriftSQL, $queryStr ) {
     $this->thriftSQL = $thriftSQL;
     $this->queryStr = $queryStr;
+
+    $this->queryCount = 1;
+    $this->doQuery();
+  }
+
+  private function doQuery() {
+    $this->buffer = array();
+    $this->location = 0;
+    $this->thriftSQLQuery = $this->thriftSQL->query( $this->queryStr )->wait();
   }
 
   public function allowRerun( $value ) {
     $this->allowRerun = (bool) $value;
     return $this;
   }
 
+  public function useSchema( $value ) {
+    $this->useSchema = (bool) $value;
+    return $this;
+  }
+
+  public function schema() {
+    return $this->thriftSQLQuery->schema();
+  }
+
   /**
-   * Return the current element
-   * @link http://php.net/manual/en/iterator.current.php
-   * @return mixed Can return any type.
-   * @since 5.0.0
```

Comment:
Why remove heredocs here and below?

Reply:
These docs are from PHP's document and are not suitable for this class. eg: `@since 5.0.0` is misleading (ThriftSQL's version is just 0.3.x)